### PR TITLE
Front end random game

### LINKED ISSUE #8: Front end to display random game

### PR DESCRIPTION
**Pull request by tjforryan**
- Added new modal where you can see a random game.  Shows up in front of form on the main page, when submitted

- "Next Random Game" will fire the same request again

RandomGameModal

- Replaced `uglify` with `terser`, as `uglify` can't support hooks yet

- Github's diff is coming out a bit unhelpful (I suspect because of the line-ending thing), so I'll add some comments to make it easier to parse

closes #8

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,129 +1,131 @@
...
-server/dist
-server/index.html
-
+
+# Created by https://www.gitignore.io/api/node,visualstudiocode,windows
+# Edit at https://www.gitignore.io/?templates=node,visualstudiocode,windows
+
+### IntelliJ ###
+.idea/
+how-many-meeples.iml
```
Comment (tjforryan) [inline .gitignore]: The above lines are all I changed here

File: client/index.js [hunk 2]
```diff
@@ -1,25 +1,27 @@
...
+    }
+});
+
+ReactDOM.render(
+    <MuiThemeProvider theme={theme}>
+        <CssBaseline />
+            <BrowserRouter>
+                <div>
+                    <Route component={GameForm}/>
+                    <Route path="/random/:listType/:listOption" component={RandomGameModal} />
```
Comment (tjforryan) [inline client/index.js]: Added the modal here and made the Game Form show up on all routes (to stay behind when the modal appears)

File: webpack.common.js [hunk 3]
```diff
@@ -1,45 +1,48 @@
...
+            },
+            {
+                test: /\.css$/,
+                use: [ MiniCssExtractPlugin.loader,'css-loader']
+            }
+        ]
+    },
+    plugins: [
+        new CleanWebpackPlugin({
+            cleanOnceBeforeBuildPatterns: ['server/dist']
```
Comment (tjforryan) [inline webpack.common.js]: API for this changed when I updated the dep

File: webpack.dev.js [hunk 4]
```diff
@@ -1,14 +1,26 @@
...
+const apiMocker = require('connect-api-mocker');
+
+module.exports = merge(common, {
+    mode: 'development',
+    devtool: 'source-map',
+    devServer: {
+        contentBase: path.join(__dirname, 'dist'),
+        port: 5000,
+
+        // Will serve index.html, if the file requested doesn't exist.
```
Comment (tjforryan) [inline webpack.dev.js]: Improved dev experience, by getting the dev server to:
- hit the server (which needs to run separately btw)
- return with the app bundle etc for routes that don't do anything else

File: client/index.js [hunk 5]
```diff
@@ -1,25 +1,27 @@
...
+import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';
+import CssBaseline from '@material-ui/core/CssBaseline';
+import GameForm from './GameForm';
+import RandomGameModal from './RandomGameModal';
+import React from 'react';
+import ReactDOM  from 'react-dom';
+
+const theme = createMuiTheme({
+    typography: {
+        fontFamily: ['SkyTextRegular', 'Verdana', 'sans-serif'],
```
Comment (Teqqles) [inline client/index.js]: SkyTextRegular? :trollface:

File: webpack.common.js [hunk 6]
```diff
@@ -1,45 +1,48 @@
...
+    output: {
+        filename: 'dist/client.js',
+        path: path.resolve(__dirname, 'server'),
+        publicPath: '/'
+    },
+    module: {
+        rules: [
+            {
+                enforce: 'pre',
+                test: /\.(js|jsx)$/,
```
Comment (Teqqles) [inline webpack.common.js]: This regex can be simplified

```suggestion
                test: /\.(jsx?)$/,
```

File: webpack.common.js [hunk 7]
```diff
@@ -1,45 +1,48 @@
...
+                exclude: /node_modules/,
+                loader: 'eslint-loader',
+                options: {
+                    fix: true
+                }
+            },
+            {
+                test: /\.(js|jsx)$/,
```
Comment (Teqqles) [inline webpack.common.js]:
```suggestion
                test: /\.(jsx?)$/,
```

**Comment by tjforryan** [comment]
@Teqqles Done!

**Pull request merged**